src/prepared_geometry.rs

```diff
@@ -21,11 +23,13 @@ use geos_sys::*;
 ///
 /// assert_eq!(prepared_geom.contains(&geom2), Ok(true));
 /// ```
-pub struct PreparedGeometry {
+pub struct PreparedGeometry<'a> {
     ptr: PtrWrap<*const GEOSPreparedGeometry>,
+    geom: Option<Geometry>,
```

Comment:
What is this field used for?

Reply:
That's so that `PreparedGeometry` can optionally own the related `Geometry` (and have optionally `static` lifetime). It either has `static` lifetime and owns the geometry, or has limited lifetime tied to external `Geometry` (in which case the `geom` field is `None`).